config/store.ts

```diff
@@ -0,0 +1,21 @@
+// config/store.js
+
+export interface StoreFile {
+  file: string; 
+  default: any
+}
+
+export interface User {
+  username: string,
+  passwordHash: string,
+  token?: string
+}
+
+const store: { [key: string]: StoreFile } = {
+  settings : { file: 'settings.json', default: {} },
+  scenes : { file: 'scenes.json', default: [] },
+  nodes : { file: 'nodes.json', default: [] },
+  users : { file: 'users.json', default: [] }
+}
```

Comment:
There's really no nice way to achieve what you're doing here.
`{ [key: string]: StoreFile }` is better written as the type `Record<string, StoreFile>`, but this does not prevent you from accessing non-existant keys. You can hard-code the shape of an object using `as const`, but that does not let you say "the value is a StoreFile".

I'd probably do this:
```
type StoreKeys = "settings" | "scenes" | "nodes" | "users";

const store: Record<StoreKeys, StoreFile> = {
  settings : { file: 'settings.json', default: {} },
  scenes : { file: 'scenes.json', default: [] },
  nodes : { file: 'nodes.json', default: [] },
  users : { file: 'users.json', default: [] },
}
```

There is another trick which involves a one-time validation function and corresponding type declaration, but that's overkill here.

Reply:
Done! Thanks for the tip :)